ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -102,17 +102,20 @@ static EngineExecutor create(
     return new EngineExecutor(engineContext, serviceContext, ksqlConfig, overriddenProperties);
   }
 
-  @SuppressWarnings("OptionalGetWithoutIsPresent") // Known to be non-empty
   ExecuteResult execute(final KsqlPlan plan) {
-    final Optional<String> ddlResult = plan.getDdlCommand()
-        .map(ddl -> executeDdl(ddl, plan.getStatementText(), plan.getQueryPlan().isPresent()));
-
-    final Optional<PersistentQueryMetadata> queryMetadata = plan.getQueryPlan()
-        .map(qp -> executePersistentQuery(qp, plan.getStatementText()));
+    if (!plan.getQueryPlan().isPresent()) {
```

Comment:
this is just a minor refactor that will help me in the next PR, but since the next one is bigger figured I'd just add it here